impl/src/test/java/org/pentaho/di/trans/dataservice/ui/controller/DataServiceTestControllerTest.java

```diff
@@ -275,6 +268,10 @@ public void testInitAssertsFail() throws Exception {
     when( document.getElementById( "window-size-row-unit" ) ).thenReturn( xulLabel );
     when( document.getElementById( "window-every-row-unit" ) ).thenReturn( xulLabel );
     when( document.getElementById( "window-limit-row-unit" ) ).thenReturn( xulLabel );
+    when( document.getElementById( "preview-opt-btn" ) ).thenReturn( xulButton );
+    when( document.getElementById( "exec-sql-btn" ) ).thenReturn( xulButton );
+    when( document.getElementById( "error-alert" ) ).thenReturn( xulLabel );
+    when( document.getElementById( "optimization-impact-info" ) ).thenReturn( xulTextBox );
```

Comment:
These additional elements are needed in a few places. I'm honestly not really sure how they weren't breaking with the older mocking behavior -- somehow the other mocks were getting hit for these...